doc/modules/neural_networks_supervised.rst

```diff
@@ -0,0 +1,357 @@
+
+The advantages of Multi-layer Perceptron are:
+
+    + Capability to learn complex/non-linear models.
+
+    + Capability to learn models in real-time (on-line learning) 
+      using ``partial_fit``.
+      
+      
+The disadvantages of Multi-layer Perceptron (MLP) include:
+
+    + MLP with hidden layers have a non-convex loss function where there exists 
+      more than one local minimum. Therefore different random weight 
+      initializations can lead to different validation accuracy.
+
+    + MLP suffers from the Backpropagation diffusion problem; layers far from 
+      the output update with decreasing momentum, leading to slow convergence.
+
+    + MLP requires tuning a number of hyperparameters such as the number of 
+      hidden neurons, layers, and iterations.
+
+    + MLP is sensitive to feature scaling.
+
+Please see :ref:`Tips on Practical Use <mlp_tips>` section that addresses 
+some of these disadvantages.
+
+
+Classification
+==============
+
+Class :class:`MLPClassifier` implements  
+a multi layer perceptron (MLP) algorithm that trains using Backpropagation. 
+
+MLP trains on two arrays: array X of size (n_samples, n_features), which holds 
+the training samples represented as floating point feature vectors; and array 
+y of size (n_samples,), which holds the target values (class labels) for the 
+training samples::
+
+    >>> from sklearn.neural_network import MLPClassifier
+    >>> X = [[0., 0.], [1., 1.]]
+    >>> y = [0, 1]
+    >>> clf = MLPClassifier(algorithm='l-bfgs', alpha=1e-5, hidden_layer_sizes=(5, 2), random_state=1)
+    >>> clf.fit(X, y) # doctest: +NORMALIZE_WHITESPACE
+    MLPClassifier(activation='relu', algorithm='l-bfgs', alpha=1e-05,
+           batch_size=200, beta_1=0.9, beta_2=0.999, early_stopping=False,
+           epsilon=1e-08, hidden_layer_sizes=(5, 2), learning_rate='constant',
+           learning_rate_init=0.001, max_iter=200, momentum=0.9,
+           nesterovs_momentum=True, power_t=0.5, random_state=1, shuffle=True,
+           tol=0.0001, validation_fraction=0.1, verbose=False,
+           warm_start=False)
+
+After fitting (training), the model can predict labels for new samples::
+
+    >>> clf.predict([[2., 2.], [-1., -2.]]) 
+    array([1, 0])
+
+MLP can fit a non-linear model to the training data. ``clf.coefs_`` 
+contains the weight matrices that constitute the model parameters::
+
+    >>> [coef.shape for coef in clf.coefs_]
+    [(2, 5), (5, 2), (2, 1)]
+
+To get the raw values before applying the output activation function, run the
+following command,
+
+use :meth:`MLPClassifier.decision_function`::
+
+    >>> clf.decision_function([[2., 2.], [1., 2.]])  # doctest: +ELLIPSIS
+    array([ 47.6...,  47.6...])
+
+Currently, :class:`MLPClassifier` supports only the 
+Cross-Entropy loss function, which allows probability estimates by running the 
+``predict_proba`` method.
+
+MLP trains using backpropagation. For classification, it minimizes the 
```

Comment:
I'm sorry, but MLPs are not trained using backpropagation. They are trained using a form of gradient descent. The gradient is calculated by backpropagation.

It is important to state this correctly here in the docs, because it may be used as a  reference in the future. 
Since sometimes the ideas are mixed up and vague, it would be better to be explicit about it what each part means.